hack/version.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+#
+# Copyright 2014 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+#
+# (Loosely) used https://github.com/kubevirt/kubevirt/blob/master/hack/version.sh as a template
+
+# Grovels through git to return build version.
+source hack/common.sh
+
+function ovscni::version::get_version_vars() {
+    # If the source was exported through git archive, then # we likely don't
+    # have a git tree, but these magic values may be filled in.
+    if [[ '$Format:%%$' == "%" ]]; then
+        OVS_CNI_GIT_COMMIT='$Format:%H$'
+        OVS_CNI_GIT_TREE_STATE="archive"
+        # When a 'git archive' is exported, the '$Format:%D$' below will look
+        # something like 'HEAD -> release-1.8, tag: v1.8.3' where then 'tag: '
+        # can be extracted from it.
+        if [[ '$Format:%D$' =~ tag:\ (v[^ ,]+) ]]; then
+            OVS_CNI_GIT_VERSION="${BASH_REMATCH[1]}"
+        fi
+    else
+        local git=(git --work-tree "${KUBEVIRT_DIR}")
+
+        OVS_CNI_GIT_COMMIT=$("${git[@]}" rev-parse "HEAD^{commit}" 2>/dev/null)
+
+        # Check if the tree is dirty.  default to dirty
+        if git_status=$("${git[@]}" status --porcelain 2>/dev/null) && [[ -z ${git_status} ]]; then
+            OVS_CNI_GIT_TREE_STATE="clean"
+        else
+            OVS_CNI_GIT_TREE_STATE="dirty"
+        fi
+
+        OVS_CNI_GIT_VERSION=${OVS_CNI_GIT_COMMIT}
```

Comment:
Would it make sense to use tag name instead of commit id in case of we are right on a tag?

Reply:
We don't have tags though. The original file was assuming there are tags, so I significantly shortened their script and removed the assumption. Perhaps I should try to stick as close to the original and just remove the assumption though; and then we can discuss if any changes to the original would be justified.

Reply:
I want to add 0.1 tag as soon as we have this PR in, we can just use the original script. Is that ok?

Reply:
Then maybe push the tag without it and I will rework the PR, syncing closer to original.

Reply:
Ok, sounds good :)

Reply:
Done, tagged HEAD with v0.1.0